libavutil/hwcontext_vaapi.c

```diff
@@ -1788,6 +1799,60 @@ static int vaapi_device_create(AVHWDeviceContext *ctx, const char *device,
     }
 #endif
 
+#if HAVE_VAAPI_WIN32
+    if (!display && try_win32) {
+        // Try to create a display from the specified device, if any.
+        if (!device) {
+            display = vaGetDisplayWin32(NULL);
+        } else {
+            IDXGIFactory2 *pDXGIFactory = NULL;
+            IDXGIAdapter *pAdapter = NULL;
+            typedef HRESULT(WINAPI *PFN_CREATE_DXGI_FACTORY)(REFIID riid, void **ppFactory);
+#if !HAVE_UWP
+            HANDLE dxgilib = dlopen("dxgi.dll", 0);
```

Comment:
sorry for ignorance, but why dlopen here and FreeLibrary down below?

Reply:
Great catch, I'll use `dlclose` to be consistent.

Reply:
Just curious: ffmpeg does provide dlopen/dlclose wrappers on Windows?

Reply:
Yes, in `"compat/w32dlfcn.h"` as ` hwcontext_d3d11va` uses.